.github/workflows/tests.yaml

```
@@ -29,7 +29,9 @@ jobs:
           cache: poetry
       - run: poetry install
       - name: Run tests
-        run: poetry run pytest tests/${{ matrix.test }}
+        run: poetry run pytest tests/${{ matrix.test }} --cov=tests/${{ matrix.test }} --cov-report=xml
```

You sure about the cov flag? Seems weird
```suggestion
        run: poetry run pytest tests/${{ matrix.test }} --cov=runner_manager --cov-report=xml
```

---

yeah since the test suite are seperate each has its own workflow so to speak. with your suggestion the code coverage will be 0% for api if i am running the test on `tests/unit` for example

---

The goal is not to see the coverage of the tests, but the coverage of the code itself. We want to know what is covered within our code.